sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -530,14 +601,21 @@ public Symbol visitArrayComparisonExpression(ArrayComparisonExpression node, Exp
                 throw new UnsupportedFeatureException("ALL is not supported");
             }
 
-            Symbol arraySymbol = process(node.getRight(), context);
+            Expression right = node.getRight();
+            if (right instanceof SubqueryExpression) {
```

Comment:
IMHO the `infstanceof` is not necessary because the `visitSubqueryExpression` will only be called (by `process`) anyway if the parser detected a `SubqueryExpression`.